code/golang/where.go

```diff
@@ -24,14 +24,14 @@ type PartitionedArgs struct {
 
 func PartitionedArgsFromWheres(wheres []*ir.Where) (out PartitionedArgs) {
 	for _, where := range wheres {
-		if where.Right != nil {
+		if !where.Right.HasPlaceholder() {
 			continue
 		}
 
 		arg := ArgFromWhere(where)
 		out.AllArgs = append(out.AllArgs, arg)
 
-		if where.Nullable() {
+		if where.NeedsCondition() {
```

Comment:
it must also be the case that if the arg is nullable, it would render as a condition, which i don't think is true. maybe the semantics are fine because the partitionedargs NullableArgs entries really mean ConditionedArgs. if that's the case, can we make that change?